src/crypto/Ed25519PrivateKey.ts

```diff
@@ -204,4 +217,32 @@ export class Ed25519PrivateKey {
     public toKeystore(passphrase: string): Promise<Uint8Array> {
         return createKeystore(this._keyData, passphrase);
     }
+
+    /**
+     * Recover a private key from a .pem file
+     *
+     * This pem method assumes the .pem file has been converted to a Uint8Array already
+     *
+     * Ensures the words "Private Key" are in the file, then looks for the next new line, slices the key from the array, then decodes and prepares it
+     */
+    public static fromPem(pemArray: Uint8Array): Ed25519PrivateKey {
+        const index = findSubarray(pemArray, privateKeyUint8Array);
+
+        if (index === -1) {
+            throw new BadPemFileError();
+        }
+
+        const newPemArray = pemArray.slice(index);
+        const firstNLIndex = newPemArray.findIndex((el) => el === 10);
```

Comment:
Perhaps use `Uint8Array.indexOf()` method instead of `Uint8Array.findIndex()` because you're looking for a specific value.

Reply:
no need, our findSubarrray refactor and fixing the subarray definitions removes the need to find Indexes